equistore-core/tests/utils/mod.rs

```diff
@@ -4,13 +4,24 @@
 use std::path::Path;
 use std::process::Command;
 
-pub fn cmake_config(source_dir: &Path, build_dir: &Path, build_type: &str) -> Command {
+fn build_type() -> &'static str {
+    // assume that debug assertion means that we are building the code in
+    // debug mode, even if that could be not true in some cases
```

Comment:
no, the comment means the reverse of what you wrote =). We would like debug mode to mean "not optimization", but I don't think we can detect that. So instead we check for "does the code also run "debug" checks/assertions" (i.e. `debug_assert!` in the code), which could be also turned on for some release/with optimization builds

Reply:
I thought you mean with that comment that sometimes you can run debug mode with some kind of optimization, so the `debug_assert` is not correctly detecting the debug mode. But since this is not the case, then I don't understand the part `even if that could be not true in some cases` in the comment. The `debug_assert` does exactly mean not optimized.

Reply:
no, `debug_assert` just mean "add some more internal consistency checks". Optimizations (inlining, loop unrolling, constant propagations, and other things LLVM does) are controlled by a separate `opt-level` setting, which I don't think we can query from inside the code.

Reply:
I don't fully get it

> debug_assert! statements are only enabled in non optimized builds by default. An optimized build will not execute debug_assert! statements unless -C debug-assertions is passed to the compiler

https://doc.rust-lang.org/std/macro.debug_assert.html

What kind of optimization it can then detect?